rotkehlchen/chain/evm/decoding/zerox/decoder.py

```diff
@@ -41,37 +41,53 @@ def __init__(
             msg_aggregator: 'MessagesAggregator',
             router_address: ChecksumEvmAddress,
             flash_wallet_address: ChecksumEvmAddress,
+            settler_routers_addresses: list[ChecksumEvmAddress] | None = None,
     ) -> None:
         """router_address is the main point of contact with the 0x protocol.
         flash_wallet_address is a contract that can execute arbitrary calls from 0x router_address.
         Docs: https://0x.org/docs/introduction/0x-cheat-sheet#exchange-proxy-addresses"""
         super().__init__(evm_inquirer, base_tools, msg_aggregator)
         self.evm_txns = EvmTransactions(self.evm_inquirer, self.base.database)
         self.router_address = router_address
+        self.settler_routers_addresses = settler_routers_addresses
         self.flash_wallet_address = flash_wallet_address
 
     def _update_send_receive_fee_events(
             self,
             send_event: 'EvmEvent | None' = None,
             receive_event: 'EvmEvent | None' = None,
             fee_event: 'EvmEvent | None' = None,
+            router_address: ChecksumEvmAddress | None = None,
     ) -> None:
         """An auxiliary function to update the send, receive and/or fee events with the 0x values"""  # noqa: E501
+        # This is a case for swaps made via the settler
+        # The send event contains the settler address, so need to
+        # update the others events with the settler address
+        if (
+            send_event is not None and
+            self.settler_routers_addresses is not None and
+            send_event.address in self.settler_routers_addresses
+        ):
+            router_address = send_event.address
+
+        if router_address is None:
```

Comment:
```suggestion
            router_address = send_event.address
        elif router_address is None:
```
As I mentioned, this should be an `elif` right?